doc/source/working_with_kiwi/xml_description.rst

```diff
@@ -0,0 +1,763 @@
+     </repository>
+     <packages type="bootstrap">
+       <package name="udev"/>
+       <package name="filesystem"/>
+       <package name="openSUSE-release"/>
+       <!-- additional packages installed before the chroot is created -->
+     </packages>
+     <packages type="image">
+       <package name="patterns-openSUSE-base"/>
+       <!-- additional packages to be installed into the chroot -->
+     </packages>
+   </image>
+
+The `packages` element provides a collection of different child elements
+that instruct KIWI when and how to perform package installation or
+removal. Each `packages` element acts as a group, whose behavior can be
+configured via the following attributes:
+
+- `type`: either `bootstrap`, `image`, `delete` or a build type (see:
+  :ref:`xml-description-build-types`). Packages for `type="bootstrap"` are
+  pre-installed to populate the images' root file system before chrooting
+  into it. Packages in `type="image"` are installed immediately after the
+  initial chroot into the new root file system. Packages in `type="delete"`
+  are removed after the *Prepare operation*. And packages which belong to a
+  build type are only installed when that specific build type is currently
+  processed by KIWI.
+
+- `profiles`: a list of profiles to which this package selection applies
+  (see :ref:`xml-description-image-profiles`).
+
+- `patternType`: selection type for patterns, supported values are:
+  `onlyRequired`, `plusRecommended`, see:
+  :ref:`xml-description-product-and-namedCollection-element`.
+
+We will describe the different child elements of `packages` in the following
+sections.
+
+.. _xml-description-package-element:
+
+The `package` element
+^^^^^^^^^^^^^^^^^^^^^
+
+The `package` element represents a single package to be installed (or
+removed), whose name is specified via the mandatory `name` attribute:
+
+.. code-block:: xml
+
+   <image schemaversion="7.0" name="JeOS-Tumbleweed">
+     <!-- snip -->
+     <packages type="bootstrap">
+       <package name="udev"/>
+     </packages>
+   </image>
+
+which adds the package `udev` to the list of packages to be added to the
+initial filesystem.
+
+Packages can also be included only on specific architectures via the `arch`
+attribute. KIWI compares the `arch` attributes value with the output of
+`uname -m`.
+
+.. code-block:: xml
+
+   <image schemaversion="7.0" name="JeOS-Tumbleweed">
+     <!-- snip -->
+     <packages type="image">
+       <package name="grub2"/>
+       <package name="grub2-x86_64-efi" arch="x86_64"/>
+       <package name="shim" arch="x86_64"/>
+     </packages>
+   </image>
+
+which results in `grub2-x86_64-efi` and `shim` being only installed on 64
+Bit images, but GRUB2 also on 32 Bit images.
+
+
+The `ignore` element
+^^^^^^^^^^^^^^^^^^^^
+
+Packages can be explicitly ignored inside a `packages` collection, e.g. to
+not remove certain dependencies in `<packages type="delete">`. This is
```

Comment:
`<packages type="delete">` does not remove dependencies. Should use `<packages type="uninstall">` instead. It is relevant to note the `<ingore>` element is applied before or meanwhile the image packages installation, thus using ignore to avoid the deletion of some package during the process of `<packages type="uninstall">` is not obvious (e.g. I doubt it could be done with DNF).

I'd say the most obvious use case of this element is in conjunction with `patternType="plusRecommended"`, this way the user can avoid to install some specific recommended package, but install all other recommendations.

Reply:
Oh, right, thanks for the pointer! I totally forgot to check the schema documentation which types are supported and only copied these from the legacy-kiwi docs. I'll change that!